jerry-ext/include/jerryscript-ext/autorelease.impl.h

```diff
@@ -0,0 +1,47 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JERRYX_AUTORELEASE_IMPL_H
+#define JERRYX_AUTORELEASE_IMPL_H
+
+#include "jerryscript.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif /* __cplusplus */
+
+/*
+ *
+ */
+#ifdef __GNUC__
+/*
+ * Calls jerry_release_value(*value).
+ * The GCC __cleanup__ function must take a pointer to the variable to clean up.
+ */
+void jerryx_autorelease (const jerry_value_t *value);
+#define __JERRYX_VAL_IMPL const jerry_value_t __attribute__ ((__cleanup__(jerryx_autorelease)))
+
+#define __JERRYX_NEXT_UNUSED_NAME __jerryx_unused_##__COUNTER__
+#define __JERRYX_UNUSED_IMPL __JERRYX_VAL_IMPL __attribute__ ((unused)) __JERRYX_NEXT_UNUSED_NAME
+#else /* !__GNUC__ */
```

Comment:
if clang supports the attribute, we should have it in the extension from the start